engine/src/main/java/org/terasology/config/flexible/FlexibleConfigManagerImpl.java

```diff
@@ -0,0 +1,151 @@
+            throw new RuntimeException("Attempting to add another config with the URI " + flexibleConfigUri);
+        }
+
+        flexibleConfigs.put(flexibleConfigUri, flexibleConfig);
+    }
+
+    @Override
+    public void removeFlexibleConfig(SimpleUri flexibleConfigUri) {
+        if (!flexibleConfigs.containsKey(flexibleConfigUri)) {
+            throw new RuntimeException("Attempting to delete non-existent config with the URI " + flexibleConfigUri);
+        }
+
+    }
+
+    @Override
+    public FlexibleConfig getFlexibleConfig(SimpleUri flexibleConfigUri) {
+        return flexibleConfigs.get(flexibleConfigUri);
+    }
+
+    @Override
+    public void load() {
+        for (Entry<SimpleUri, FlexibleConfig> entry : flexibleConfigs.entrySet()) {
+            SimpleUri flexibleConfigUri = entry.getKey();
+            FlexibleConfig flexibleConfig = entry.getValue();
+
+            try (Reader reader = Files.newBufferedReader(getPathForFlexibleConfig(flexibleConfigUri), TerasologyConstants.CHARSET)) {
+                JsonReader jsonReader = new JsonReader(reader);
+
+                Map<SimpleUri, String> unusedSettings = Maps.newHashMap();
+
+                if (jsonReader.hasNext()) {
+                    jsonReader.beginObject();
+
+                    while (jsonReader.hasNext()) {
+                        SimpleUri id = new SimpleUri(jsonReader.nextName());
+                        String value = jsonReader.nextString();
+                        unusedSettings.put(id, value);
+                    }
+
+                    jsonReader.endObject();
+                }
+
+                flexibleConfig.setUnusedSettings(unusedSettings);
+            } catch (IOException e) {
+                logger.error("Failed to load config file {}, falling back on default config");
+            }
+        }
+    }
+
+    @Override
+    public void save() {
+        for (Entry<SimpleUri, FlexibleConfig> entry : flexibleConfigs.entrySet()) {
+            SimpleUri flexibleConfigUri = entry.getKey();
+            FlexibleConfig flexibleConfig = entry.getValue();
+
+            try (BufferedWriter writer = Files.newBufferedWriter(getPathForFlexibleConfig(flexibleConfigUri), TerasologyConstants.CHARSET)) {
+                JsonObject jsonObject = new JsonObject();
+
+                Map<SimpleUri, Setting> settingMap = flexibleConfig.getActiveSettings();
+                Map<SimpleUri, String> unusedSettings = flexibleConfig.getUnusedSettings();
+
+                for (Entry<SimpleUri, Setting> activeSetting : settingMap.entrySet()) {
+                    Setting setting = activeSetting.getValue();
+                    if (!setting.getValue().equals(setting.getDefaultValue())) {
+                        jsonObject.addProperty(activeSetting.getKey().toString(), setting.getValue().toString());
+                    }
+                }
+
+                // Add all the non-default settings that were not used in this session
```

Comment:
This is like old save files - we can't delete them based on suspicion of being a zombie, since they might very well be a vampire instead.

I think we should leave this untouched, since I'd rather live with 10 extra settings than lose all my settings because I did not load a module for (say) 10 consecutive games.

Reply:
Hmmmm. I wonder if only the engine should add settings to an engine config and only modules should add settings to their very own module config. This should eliminate zombie settings as configs are atomically loaded and atomically discarded.

Programmatically this might not have consequences, but when it's time to group settings, potentially from different configs, into a single UI, that's where the uiPath idea would become a necessity. 

What do you think?

Reply:
> This should eliminate zombie settings as configs are atomically loaded and atomically discarded.
I'm sure I'm missing something, but I can't see how different config files can prevent zombie settings from piling up. (The only possible way I can see is deleting config associated with myModule when user deletes myModule, but that's not a normal scenario, since I don't expect user to be deleting modules all the time.)